cylc/flow/task_events_mgr.py

```diff
@@ -1536,10 +1536,11 @@ def _insert_task_job(
             job_conf = itask.jobs[-1]
 
         # insert job into data store
+        job_status = 'submitted' if submit_status == 0 else 'submit-failed'
         self.data_store_mgr.insert_job(
```

Comment:
This method takes a job status, not a task status. If the task status is 'waiting' it returns None and doesn't register the job with the data store.